src/IonTextWriter.ts

```diff
@@ -328,6 +328,9 @@ export class TextWriter extends AbstractWriter {
     protected writeAnnotations() : void {
         if (this._annotations === null || this._annotations === undefined) return;
         for (let annotation of this._annotations) {
+            if (typeof annotation !== 'string') {
```

Comment:
don't you need to check that `_annotations` is an array?

I believe it'd be better to do this check where `_annotations` array is being mutated. This way you fail fast when it's initialized by anything that's not an array or when trying to  add an element that's not a string.

Reply:
in the setAnnotations call yes we do.